3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/OlarkAdapter.scala

```diff
@@ -0,0 +1,195 @@
+   *         Success, or a NEL of Failure Strings
+   */
+  def toRawEvents(payload: CollectorPayload)(implicit resolver: Resolver): ValidatedRawEvents = {
+    (payload.body, payload.contentType) match {
+      case (None, _)                          => s"Request body is empty: no ${VendorName} events to process".failNel
+      case (_, None)                          => s"Request body provided but content type empty, expected ${ContentType} for ${VendorName}".failNel
+      case (_, Some(ct)) if ct != ContentType => s"Content type of ${ct} provided, expected ${ContentType} for ${VendorName}".failNel
+      case (Some(body),_) if (body.isEmpty)   => s"${VendorName} event body is empty: nothing to process".failNel
+      case (Some(body),_)                     => {
+        try {
+          val qsParams = toMap(payload.querystring)
+          val bodyMap = toMap(URLEncodedUtils.parse(URI.create("http://localhost/?" + body), "UTF-8").toList)
+          (payloadBodyToEvent(bodyMap)) match {
+            case (Failure(err))    => err.failNel
+            case (Success(event))  => {
+              val hasOperators = (event \ "operators")
+              val eventType = hasOperators match {
+                case (JNothing) => Some("offline_message")
+                case (_)        => Some("transcript")
+              }
+              lookupSchema(eventType, VendorName, EventSchemaMap) match {
+                case Failure(str)    => str.fail
+                case Success(schema) => {
+                  val transformedEvent = transformTimestamps(event)
+                  NonEmptyList(RawEvent(
+                    api          = payload.api,
+                    parameters   = toUnstructEventParams(TrackerVersion, qsParams, schema, camelize(transformedEvent), "srv"),
+                    contentType  = payload.contentType,
+                    source       = payload.source,
+                    context      = payload.context
+                  )).success
+                }
+              }
+            }
+          }
+        } catch {
+          case NonFatal(e) => s"${VendorName} could not parse body: [${JU.stripInstanceEtc(e.getMessage).orNull}]".failNel
+        }
+      }
+    }
+  }
+  /**
+   * Converts all olark timestamps in a parsed transcript or offline_message json object to iso8601 strings
+   *
+   * @param json a parsed event
+   * @return JObject the event with timstamps replaced
+   */
+  private def transformTimestamps(json: JValue): JValue = {
+    json.transformField {
+      case JField("items", jArray) => ("items", jArray.transform {
+        case jo: JObject => jo.transformField {
+          case JField("timestamp", JString(value)) => ("timestamp", toIso8601(toMsec(value)))
+        }
+      })
+    }
+  }
+  /**
+   * Converts an Olark timestamp (e.g. 1307116657.1) to msec
+   *
+   * @param oTs olark timestamp string
+   * @return Long msec
+   * @throws MatchError and NumberFormatException
```

Comment:
pls handle those at the lowest level possible, make this an option or a Validation

It shouldn't be the caller's responsibility to blindly handle a function's shortcomings.